src/sentry/snuba/outcomes.py

```diff
@@ -0,0 +1,29 @@
+from sentry.utils.snuba import (
+    raw_query,
+)
+from sentry.snuba.dataset import Dataset
+from sentry_relay import DataCategory
+
+
+def query(groupby, start, end, rollup, aggregations, orderby, fields=None, filter_keys=None):
+
+    # TODO: if roll up >= use hourly, else use raw
```

Comment:
You might also want to move the rollup constraint here instead of in the endpoint. That will help with not making invalid queries if this module is used elsewhere in the future.